src/Microsoft.Azure.SignalR/ServiceOptionsSetup.cs

```diff
@@ -10,31 +10,53 @@ namespace Microsoft.Azure.SignalR
 {
     internal class ServiceOptionsSetup : IConfigureOptions<ServiceOptions>
     {
+        private readonly string _appName;
         private readonly string _connectionString;
         private readonly ServiceEndpoint[] _endpoints;
 
         public ServiceOptionsSetup(IConfiguration configuration)
         {
-            var (connectionString, endpoints) = GetEndpoint(configuration, Constants.ConnectionStringDefaultKey, Constants.ConnectionStringKeyPrefix);
+            _appName = GetAppName(configuration);
+
+            var (connectionString, endpoints) = GetEndpoint(configuration, Constants.ConnectionStringDefaultKey, Constants.ConnectionStringKeyPrefix, _appName);
 
             // Fallback to ConnectionStrings:Azure:SignalR:ConnectionString format when the default one is not available
             if (endpoints.Count == 0)
             {
-                (connectionString, endpoints) = GetEndpoint(configuration, Constants.ConnectionStringSecondaryKey, Constants.ConnectionStringSecondaryKeyPrefix);
+                (connectionString, endpoints) = GetEndpoint(configuration, Constants.ConnectionStringSecondaryKey, Constants.ConnectionStringSecondaryKeyPrefix, _appName);
             }
 
             _connectionString = connectionString;
             _endpoints = endpoints.ToArray();
         }
 
+        private string GetAppName(IConfiguration configuration)
+        {
+            foreach (var pair in configuration.AsEnumerable())
```

Comment:
simply use `configuration[Constants.ApplicationNameDefaultKey]` and no need to support prefix?

Reply:
Sorted.